src/error.cpp

```diff
@@ -0,0 +1,14 @@
+#include "openmc/error.h"
+
+#include "openmc/message_passing.h"
+
+namespace openmc {
+
+#ifdef OPENMC_MPI
+void abort_mpi(int code)
+{
+    MPI_Abort(mpi::intracomm, code);
+}
+#endif
```

Comment:
Is there any particular reason you put this implementation in a .cpp file rather than the .h?  Trying to avoid future circular dependencies with error.h and message_passing.h?  Otherwise, I think it might make sense to just leave this short implementation in the header file.

Reply:
Yes, because this function is called from Fortran, it can't be inline. If the compiler were to optimizing away the function call, there would be no `abort_mpi` symbol for Fortran to call.